performance_tests/test/keyrings/test_aws_kms_keyring.py

```diff
@@ -0,0 +1,201 @@
+              prompt='Filename containing plaintext data you want to encrypt')
+@click.option('--kms_key_id',
+              default='arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-55776cbb2f7f')
+@click.option('--n_iters',
+              default=PerfTestUtils.DEFAULT_N_ITERS)
+@click.option('--output_file',
+              default='/'.join(__file__.split("/")[:-3]) + '/results/kms_keyring_encrypt')
+def encrypt(
+    plaintext_data_filename: str,
+    kms_key_id: str,
+    n_iters: int,
+    output_file: str
+):
+    """Performance test for the encrypt_using_keyring function."""
+    plaintext_data = PerfTestUtils.read_file(plaintext_data_filename)
+
+    keyring = create_keyring(kms_key_id)
+    time_list = []
+
+    for _ in tqdm(range(n_iters)):
+        curr_time = time.time()
+
+        encrypt_using_keyring(plaintext_data, keyring)
+
+        # calculate elapsed time in milliseconds
+        elapsed_time = (time.time() - curr_time) * 1000
+        time_list.append(elapsed_time)
+
+    PerfTestUtils.write_time_list_to_csv(time_list, output_file)
+
+
+@click.group()
+def decrypt_kms_keyring():
+    """Click group helper function"""
+
+
+@decrypt_kms_keyring.command()
+@click.option('--ciphertext_data_filename',
+              default='/'.join(__file__.split("/")[:-2]) + '/resources/ciphertext/kms/ciphertext-data-'
+              + PerfTestUtils.DEFAULT_FILE_SIZE + '.ct',
+              prompt='Filename containing ciphertext data you want to decrypt')
+@click.option('--kms_key_id',
+              default='arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-55776cbb2f7f')
+@click.option('--n_iters',
+              default=PerfTestUtils.DEFAULT_N_ITERS)
+@click.option('--output_file',
+              default='/'.join(__file__.split("/")[:-3]) + '/results/kms_keyring_decrypt')
+def decrypt(
+    ciphertext_data_filename: str,
+    kms_key_id: str,
+    n_iters: int,
+    output_file: str
+):
+    """Performance test for the decrypt_using_keyring function."""
+    ciphertext_data = PerfTestUtils.read_file(ciphertext_data_filename)
+
+    keyring = create_keyring(kms_key_id)
+    time_list = []
+
+    for _ in tqdm(range(n_iters)):
+        curr_time = time.time()
+
+        decrypt_using_keyring(ciphertext_data, keyring)
+
+        # calculate elapsed time in milliseconds
+        elapsed_time = (time.time() - curr_time) * 1000
+        time_list.append(elapsed_time)
+
+    PerfTestUtils.write_time_list_to_csv(time_list, output_file)
+
+
+kms_keyring_test = click.CommandCollection(sources=[create_kms_keyring,
+                                                    create_kms_keyring_given_kms_client,
+                                                    encrypt_kms_keyring,
+                                                    decrypt_kms_keyring])
+
+
+@pytest.fixture
+def runner():
+    """Click runner"""
+    return click.testing.CliRunner()
+
+
+def test_create(runner):
```

Comment:
[curious] how much work would it be to print out the summary statistics to the codebuild logs?

Reply:
We would be calling the consolidate results script. It would just require getting the correct absolute paths I think.